tests/flow/test_entity_update.py

```diff
@@ -134,3 +141,99 @@ def test13_invalid_complex_type_in_array(self):
                 self.env.assertTrue(False)
             except ResponseError as e:
                 self.env.assertContains("Property values can only be of primitive types or arrays of primitive types", str(e))
+
+    # fail when attempting to perform invalid map assignment
+    def test14_invalid_map_assignment(self):
+        try:
+            graph.query("MATCH (a) SET a.v = {f: true}")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Property values can only be of primitive types or arrays of primitive types", str(e))
+
+    # update properties by attribute set reassignment
+    def test15_assign_entity_properties(self):
+        # merge attribute set of a node with existing properties
+        node = Node(label="L", properties={"v1": 1, "v2": 2})
+        result = multiple_entity_graph.query("MATCH (n1 {v1: 1}), (n2 {v2: 2}) SET n1 += n2 RETURN n1")
+        expected_result = [[node]]
+        self.env.assertEqual(result.result_set, expected_result)
+        # validate index updates
+        result = multiple_entity_graph.query("MATCH (n:L) WHERE n.v1 > 0 RETURN n.v1 ORDER BY n.v1")
+        expected_result = [[1]]
+        self.env.assertEqual(result.result_set, expected_result)
+        result = multiple_entity_graph.query("MATCH (n:L) WHERE n.v2 > 0 RETURN n.v2 ORDER BY n.v2")
+        expected_result = [[2],
+                           [2]]
+        self.env.assertEqual(result.result_set, expected_result)
+
+        # overwrite attribute set of node with attribute set of edge
+        node = Node(label="L", properties={"v1": 3})
+        result = multiple_entity_graph.query("MATCH (n {v1: 1})-[e]->() SET n = e RETURN n")
+        expected_result = [[node]]
+        self.env.assertEqual(result.result_set, expected_result)
+        # validate index updates
+        result = multiple_entity_graph.query("MATCH (n:L) WHERE n.v1 > 0 RETURN n.v1 ORDER BY n.v1")
+        expected_result = [[3]]
+        self.env.assertEqual(result.result_set, expected_result)
+        result = multiple_entity_graph.query("MATCH (n:L) WHERE n.v2 > 0 RETURN n.v2 ORDER BY n.v2")
+        expected_result = [[2]]
+        self.env.assertEqual(result.result_set, expected_result)
+
+    # repeated attribute set reassignment
+    def test16_assign_entity_properties(self):
+        # repeated merges to the attribute set of a node
+        node = Node(label="L", properties={"v1": 3, "v2": 2})
+        result = multiple_entity_graph.query("MATCH (n), (x) WHERE ID(n) = 0 SET n += x RETURN n")
```

Comment:
isn't this order dependent ? wouldn't we get different values if the order of `x` was reversed ?